src/ImageSharp/Configuration.cs

```diff
@@ -1,272 +1,244 @@
+
+        /// <summary>
+        /// Gets the currently registered <see cref="IImageFormat"/>s.
+        /// </summary>
+        public IEnumerable<IImageFormat> ImageFormats => this.ImageFormatsManager.ImageFormats;
+
+        /// <summary>
+        /// Gets or sets the position in a stream to use for reading when using a seekable stream as an image data source.
+        /// </summary>
+        public ReadOrigin ReadOrigin { get; set; } = ReadOrigin.Current;
+
+        /// <summary>
+        /// Gets or sets the <see cref="ImageFormatManager"/> that is currently in use.
+        /// </summary>
+        public ImageFormatManager ImageFormatsManager { get; set; } = new ImageFormatManager();
+
+        /// <summary>
+        /// Gets or sets the <see cref="MemoryManager"/> that is currently in use.
+        /// </summary>
+        public MemoryManager MemoryManager { get; set; } = ArrayPoolMemoryManager.CreateDefault();
+
+        /// <summary>
+        /// Gets the maximum header size of all the formats.
+        /// </summary>
+        internal int MaxHeaderSize => this.ImageFormatsManager.MaxHeaderSize;
+
+        /// <summary>
+        /// Gets the currently registered <see cref="IImageFormatDetector"/>s.
+        /// </summary>
+        internal IEnumerable<IImageFormatDetector> FormatDetectors => this.ImageFormatsManager.FormatDetectors;
+
+        /// <summary>
+        /// Gets the currently registered <see cref="IImageDecoder"/>s.
+        /// </summary>
+        internal IEnumerable<KeyValuePair<IImageFormat, IImageDecoder>> ImageDecoders => this.ImageFormatsManager.ImageDecoders;
+
+        /// <summary>
+        /// Gets the currently registered <see cref="IImageEncoder"/>s.
+        /// </summary>
+        internal IEnumerable<KeyValuePair<IImageFormat, IImageEncoder>> ImageEncoders => this.ImageFormatsManager.ImageEncoders;
+
+#if !NETSTANDARD1_1
+        /// <summary>
+        /// Gets or sets the filesystem helper for accessing the local file system.
+        /// </summary>
+        internal IFileSystem FileSystem { get; set; } = new LocalFileSystem();
+#endif
+
+        /// <summary>
+        /// Gets or sets the image operations provider factory.
+        /// </summary>
+        internal IImageProcessingContextFactory ImageOperationsProvider { get; set; } = new DefaultImageOperationsProviderFactory();
+
+        /// <summary>
+        /// Registers a new format provider.
+        /// </summary>
+        /// <param name="configuration">The configuration provider to call configure on.</param>
+        public void Configure(IConfigurationModule configuration)
+        {
+            Guard.NotNull(configuration, nameof(configuration));
+            configuration.Configure(this);
+        }
+
+        /// <summary>
+        /// Registers a new format provider.
+        /// </summary>
+        /// <param name="format">The format to register as a known format.</param>
+        public void AddImageFormat(IImageFormat format)

```

Comment:
@tocsoft @JimBobSquarePants wouldn't it cleaner to have our API without these shortcuts, exposing `AddImageFormat()`, `SetEncoder()` etc. only on `ImageFormatManager`?
Or should we keep them for compatibility?

Reply:
I would be inclined to remove them.

Reply:
I didn't remove them precisely because backwards compatibility. As an intermediate solution, they can be tagged with the Obsolete attribute for a while, so developers will have a cautionary warning to point to the right methods, and they can finally be removed after a while.

Reply:
I wouldn't bother, we've broken tons of things recently and we'll be moving things about more when be fix up the namespaces.